### ISSUE TITLE
[netcore] Implement RuntimeAssembly.IsCollectible Behavior  

### ISSUE THREAD

**Issue by steveisok** [mono/mono#15142]
The property exists in RuntimeAssembly, but it needs flushed out.  

Currently:

```csharp
                public override string Location {
			get {
				return get_location ();
			}
		}

		// TODO:
		public override bool IsCollectible => false;
```

**Comment by CoffeeFlux** [comment]
Related tests to check:
```
####################################################################
##  System.Runtime.Loader.Tests
####################################################################

# NOTE: Not run (relies on collectible ALCs)
-nomethod System.Runtime.Loader.Tests.ContextualReflectionTest.*

# relies on collectible AssemblyLoadContext
-nomethod System.Runtime.Loader.Tests.AssemblyLoadContextTest.Unload_*
-nomethod System.Runtime.Loader.Tests.AssemblyLoadContextTest.Finalizer_CollectibleWithNoAssemblyLoaded
-nomethod System.Runtime.Loader.Tests.AssemblyLoadContextTest.PublicConstructor_Theory
-nomethod System.Runtime.Loader.Tests.AssemblyLoadContextTest.Unsupported_FixedAddressValueType
```

**Comment by alexischr** [comment]
This actually seems to have been completed with https://github.com/dotnet/runtime/commit/d07eade004cf38224d2d2f9a0cf51189c721e1dd , I'll see if the tests can be enabled and then we can probably close the issue

**Comment by alexischr** [comment]
The following tests are still failing:
```
ContextualReflectionTest.*
AssemblyLoadContextTest.Unload_*
AssemblyLoadContextTest.Unsupported_FixedAddressValueType
```

The other two are enabled in https://github.com/dotnet/runtime/pull/38581

**Comment by SamMonoRT** [comment]
@CoffeeFlux - this is low priority.

**Comment by srxqds** [comment]
> @CoffeeFlux - this is low priority.

why not make it work immdiately? we are can't waiting for it, as this said https://github.com/mono/mono/issues/19312
I think the alc unloadable is most import feature of .net core, but in mono runtime, it is always not work. this is so said for many product. We hope you can consider to make it work, beacuse our project will be blocked by lack of unloadable assembly.

**Comment by CoffeeFlux** [comment]
Closing in favor of https://github.com/dotnet/runtime/issues/34072
